fontforge/parsettfatt.c

```diff
@@ -1178,6 +1179,7 @@ return;
 	    if ( feof(ttf)) {
 		LogError( _("Unexpected end of file in contextual chaining subtable.\n") );
 		info->bad_ot = true;
+                free(glyphs);
```

Comment:
Seems like `rules` needs to be freed here too. Which means iterating, say, `k` from 0 to `i` (inclusive) and freeing `rules[k].subrules`.

Reply:
I noticed these structs and thought I would let this for a second pass.
Anyway, I applied what you indicated and copied it at different locations when returning because of an error